src/audio/module_adapter/module_adapter.c

```diff
@@ -240,9 +241,17 @@ static int module_adapter_dp_queue_prepare(struct comp_dev *dev)
 			 &sink_buffer->stream.runtime_stream_params,
 			 sizeof(sink_buffer->stream.runtime_stream_params));
 		/* calculate time required the module to provide OBS data portion - a period */
-		unsigned int sink_period = 1000000 * sink_get_min_free_space(mod->sinks[i]) /
+		uint32_t min_free_size =
```

Comment:
> @jxstelter , please specify which converter have problem. This is quite tricky for set sink period, @singalsu have a table to show different produce. it may not easy to set here, check below 48 -->88.2 example: ![image](https://user-images.githubusercontent.com/17277136/284199823-57a5fb0c-cce3-4747-bf14-e3e5b8085563.png)
> 
> max is 98, we may need more discuss for SRC period setting, it is more complex than we imagine.

Currently the issue is observed only on LNL FPGA setup. This is because on FPGA there is much lower MCPS margin available. In that case simply Copier->SRC->Copier pipeline is affected.  When sink_period is set for too high value, DP module get processing time later than it should and on FPGA it can not finish processing on time, causing glitches. 
The problem is not observed on RVP yet, however it could be observed when core utilization will be much higher.